src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -610,7 +609,7 @@ protected bool TryGetValue(FieldsToFetch.FieldToFetch fieldToFetch, Document doc
 
             //_loadedDocuments.Clear(); - explicitly not clearing this, we want to cache this for the duration of the query
 
-            _loadedDocuments[document.Id ?? string.Empty] = document;
+            _loadedDocuments[document.Id ?? string.Empty] = document.Clone(_context);
```

Comment:
We shouldn't clone it. For that purpose we have `IgnoreDispose`. See my above comment